agnpy/spectra/tests/test_spectra.py

```diff
@@ -817,4 +827,63 @@ def test_compare_calculation_with_calculation_split_into_two(self):
             eval_2.evaluate(gammas, 1, gamma_min, gamma_max),
             0.001)
 
+    def test_total_number_of_electrons_should_stay_after_time_evolution(self):
+        R_b = 1e16 * u.cm
+        n_e = PowerLaw.from_total_energy(
+            1e48 * u.erg,
+            4 / 3 * np.pi * R_b ** 3,
+            p=2.8,
+            gamma_min=1e2,
+            gamma_max=1e7,
+            mass=m_e,
+        )
+
+        blob = Blob(R_b, Distance(1e27, unit=u.cm).z, 10, 10, 1 * u.G, n_e=n_e)
+        synch = Synchrotron(blob)
+        ssc = SynchrotronSelfCompton(blob)
+
+        time = 60 * u.s
+        intervals = 600
+
+        electronsBefore = n_e.integrate(n_e.gamma_min, n_e.gamma_max)
+
+        blob.n_e_time_evolution(synch.electron_energy_loss_rate, time, subintervals_count=intervals)
+        electronsAfterSync = blob.n_e.integrate(blob.n_e.gamma_min, blob.n_e.gamma_max)
+
+        blob.n_e = n_e
```

Comment:
I am wondering if it is enough to only overwrite n_e, or do you need to restore the original gamma factors as well, because those are also modified in the time evolution function

Reply:
fixed, i reset the gamma params in the n_e setter